src/components/Hub/HubFilters.tsx

```diff
@@ -21,49 +15,48 @@ type HubFilterProps = {
 export const getSelectedFilters = (filters: Filter[]): FilterParams => {
   return {
     kind: getCheckedFilterValues(filters[0]),
-    keywords: getCheckedFilterValues(filters[1]),
+    keywords: [...getCheckedFilterValues(filters[1]), ...getCheckedFilterValues(filters[2]), ...getCheckedFilterValues(filters[3]),],
   }
 }
 
 export const getCheckedFilterValues = (filter: Filter) => {
-  return Object.keys(filter.values).reduce((acc, key) => {
-    let filterValue = filter.values
-    return filterValue[key] ? [...acc, key] : acc
+  return filter.values.reduce((acc, {name, selected}) => {
+    return selected ? [...acc, name] : acc
   }, [] as string[])
 }
 
 const FiltersTitle = styled.div`
   font-weight: 500;
   font-size: 1.25rem;
+  color: ${(props) => props.theme.palette.headerTextColor};
 `
-
-const FiltersContainer = styled.div`
+const FilterTitleContainer = styled.div`
   display: flex;
-  flex-direction: column;
+  justify-content: space-between;
 `
-// Hide checkbox but keep it in the DOM for accessibility
-const Checkbox = styled.input`
-  margin-right: -1rem;
-  opacity: 0;
+const ClearButton = styled(Button)`
+  color: ${props => props.theme.palette.mutedText};
 `
-type CheckboxLabelProps = {
-  checked: boolean
-  highlightColor: string
-}
-const CheckboxLabel = styled.label`
-  border: ${(props: CheckboxLabelProps) =>
-    props.checked ? `1px solid ${props.highlightColor}` : "none"};
-  border-radius: 0.25rem;
-  padding: 0.25rem 0.5rem;
+const FiltersContainer = styled.div`
```

Comment:
This is more of a question/invite for discussion, than a comment: Should we consider implementing the complete structure of MaterialUI, which would mean using Box, Grid, Container or shall we just use certain components like Buttons or DropDownMenus?

Reply:
This is a good question, we should probably agree on exactly what material components we are using.